### PR TITLE
Added support for notification LEDs and the appropriate preferences

### LINKED ISSUE #632: Can you add Samsung's API for Samsung Devices to have LED notifications?
See title. Most of the apps has the option to use the LED notification incidator on some devices like Samsung variants. Does this app have the option to do so, including changing the LED notifications colors to differ from other apps? If not, please add it in next release.

Android Version: 5.1.1
Device Variant: SM-G920A
Device Name: Galaxy S6
Manufacturer: Samsung
Carrier: AT&T

### PR DESCRIPTION
**Pull request by nwalters512**
All the color selection/preference stuff is implemented as a separate library. This could be useful outside the context of just this app; maybe we want to start breaking out some of our libraries into separate projects and publishing them to Maven Central.

Code review can start, but I'd like to wrap up some of the documentation before we merge (and maybe figure out a way to test the color library).

Fixes #632

Imgur
Imgur

---
This change is [Reviewable](https://reviewable.io/reviews/the-blue-alliance/the-blue-alliance-android/654)

### CODE REVIEW

File: colorpicker/src/main/java/com/thebluealliance/colorpicker/ColorPreference.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    private ColorPalette mColorPalette;
+    private View mColorView;
+
+    public ColorPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        TypedArray a = context.getTheme().obtainStyledAttributes(attrs, R.styleable
+                .ColorPreference, 0, 0);
+        try {
+            int id = a.getResourceId(R.styleable.ColorPreference_colors, 0);
```
Comment (nwalters512) [inline colorpicker/src/main/java/com/thebluealliance/colorpicker/ColorPreference.java]: Add a reasonable default array of colors, or throw an exception if `colors` is not present

Comment (nwalters512) [inline colorpicker/src/main/java/com/thebluealliance/colorpicker/ColorPreference.java]: Resolved in 5e4f209

**Comment by phil-lopreiato** [comment]
Reviewed 33 of 35 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[colorpicker/src/androidTest/java/com/thebluealliance/colorpicker/ApplicationTest.java, line 9 \[r2\]](https://reviewable.io:443/reviews/the-blue-alliance/the-blue-alliance-android/654#-KDz0FpkPUubTLPzejxq:-KDz0FpkPUubTLPzejxr:-56267436) ([raw file](https://github.com/the-blue-alliance/the-blue-alliance-android/blob/795198882a7d7b511ce3104affa93fac42653ee9/colorpicker/src/androidTest/java/com/thebluealliance/colorpicker/ApplicationTest.java#L9)):*
What's the purpose of this?

---

*[colorpicker/src/test/java/com/thebluealliance/colorpicker/ExampleUnitTest.java, line 10 \[r2\]](https://reviewable.io:443/reviews/the-blue-alliance/the-blue-alliance-android/654#-KDz0A-A98RiIL1PqNdn:-KDz0A-A98RiIL1PqNdo:2004122500) ([raw file](https://github.com/the-blue-alliance/the-blue-alliance-android/blob/795198882a7d7b511ce3104affa93fac42653ee9/colorpicker/src/test/java/com/thebluealliance/colorpicker/ExampleUnitTest.java#L10)):*
This doesn't seem necessary

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/the-blue-alliance/the-blue-alliance-android/654)*

**Comment by nwalters512** [comment]
The test files are auto-generated by Android Studio when you create a new library module. I left them in there in case I did end up writing tests.

**Comment by nwalters512** [comment]
All the color selection stuff has been broken out into https://github.com/the-blue-alliance/spectrum

**Comment by phil-lopreiato** [comment]
Reviewed 6 of 33 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/the-blue-alliance/the-blue-alliance-android/654)*

**Pull request merged**